### DISCUSSION TITLE
Equivalence of ws.readyState == WebSocket.OPEN ?

### DISCUSSION THREAD

**Discussion by ohenepee** [uNetworking/uWebSockets.js#149] [Q&A] [answered]
Is there anyway to know if the client is still connected or if the client websocket is still open?

**Comment by ghost** [comment]
The connection is **considered** open until close event fires. This is determined by the idleTimeout. Keep sockets alive by having the client send phony messages (messages with no real content) from time to time.

**Comment by Cheddah** [comment]
@alexhultman Conversely, as soon as a close event fires, is the **connection already closed** (ws it not waiting for an acknowledgement from the client)? Thank you!

**Comment by ghost** [comment, marked as answer]
Yes if you get close event it is closed, that's in the docs
